### PR TITLE
[Remove discover] Implement embeddable dashboard on Threat Hunting module

### PR DESCRIPTION
**Pull request by jbiset**
## Description
Implement the embeddable dashboard on Threat hunting -> dashboard tab and deprecate any use of kibana-integrations components.
 
## Issues Resolved
- #6478 

## Evidence

- [x] Each visualization, if applicable, must have interaction so that it adds the corresponding filter(s) upon clicking.

[Evidence_1.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/c89dd780-0679-45bd-add3-044e730e1c99)

- [x] The visualizations have to be updated according to the filters applied in the searchbar.

[Evidence_2.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/105f7a6e-82d0-451a-9f92-c870d80d0bcc)

- [x] The visualizations have to be updated when a search is performed in the searchbar.

[Evidence_3.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/7c8f9d5b-469b-4b80-9222-dd89533b39e9)

- [x] If there are no results, the corresponding message must appear that there are no results and the visualizations should not be rendered.

image

- [x] If there is SampleData, the corresponding SampleData message must appear.

image

- [x] If an agent is pinned, the views must be updated and changed to the agent view, if applicable.

[Evidence_6.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/756c28d9-ea70-4bf5-bb22-db82181dbdad)

- [x] Check filter behavior with respect to hide alerts, allow agents and filter order.

image

image

## Test
Go to Threats Hunting and check the following:
- [ ] Each visualization, if applicable, must have interaction so that it adds the corresponding filter(s) upon clicking.
- [ ] The visualizations have to be updated according to the filters applied in the searchbar.
- [ ] The visualizations have to be updated when a search is performed in the searchbar.
- [ ] If there are no results, the corresponding message must appear that there are no results and the visualizations should not be rendered.
- [ ] If there is SampleData, the corresponding SampleData message must appear.
- [ ] If an agent is pinned, the views must be updated and changed to the agent view, if applicable. 
- [ ] Check filter behavior with respect to `hide alerts`, `allow agents` and filter order. 

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: plugins/main/public/components/common/hocs/withPinnedAgent.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+
+import React from 'react';
+import { ModulesHelper } from '../modules/modules-helper';
+
+export const withPinnedAgent = WrappedComponent => props => {
+  const pinnedAgent = ModulesHelper.getImplicitPinnedAgent('wazuh-alerts-*');
```
Comment (Desvelao) [inline plugins/main/public/components/common/hocs/withPinnedAgent.tsx:17]: **issue**: the alerts index pattern depends on the selection of the user.

File: plugins/main/public/components/overview/threat-hunting/dashboard/dashboard.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+interface DashboardThreatHuntingProps {
+  pinnedAgent: Filter;
+}
+
+export const DashboardThreatHunting: React.FC<DashboardThreatHuntingProps> = ({
+  pinnedAgent,
+}) => {
+  /* TODO: Analyze whether to use the new index pattern handler https://github.com/wazuh/wazuh-dashboard-plugins/issues/6434 */
+  const TH_INDEX_PATTERN_ID = WAZUH_ALERTS_PATTERN;
```
Comment (Desvelao) [inline plugins/main/public/components/overview/threat-hunting/dashboard/dashboard.tsx:25]: **issue**: the alerts index pattern depends on the selection of the user.

File: plugins/main/public/components/overview/threat-hunting/dashboard/dashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            />
+          </div>
+        )}
+        {!isDataSourceLoading && dataSource && results?.hits?.total > 0 ? (
+          <SampleDataWarning />
+        ) : null}
+        {!isDataSourceLoading && dataSource && results?.hits?.total === 0 ? (
+          <DiscoverNoResults />
+        ) : null}
+        {!isDataSourceLoading && dataSource && results?.hits?.total > 0 ? (
```
[the remark this answers is not in the document]
Comment (jbiset) [reply plugins/main/public/components/overview/threat-hunting/dashboard/dashboard.tsx:207]: Improved condition for rendering the dashboard and SampleData message

File: plugins/main/public/components/common/modules/modules-defaults.tsx [hunk 4]
```diff
@@ -44,7 +45,10 @@ import { mitreAttackColumns } from '../../overview/mitre/events/mitre-attack-col
 import { virustotalColumns } from '../../overview/virustotal/events/virustotal-columns';
 import { malwareDetectionColumns } from '../../overview/malware-detection/events/malware-detection-columns';
 import { WAZUH_VULNERABILITIES_PATTERN } from '../../../../common/constants';
-import { AlertsVulnerabilitiesDataSource } from '../data-source';
+import {
+  AlertsDataSource,
```
Comment (yenienserrano) [inline plugins/main/public/components/common/modules/modules-defaults.tsx:49]: Is it necessary to create the alertsThreatHuntingDataSource?

File: plugins/main/public/components/common/modules/modules-helper.js [hunk 5]
```diff
@@ -182,4 +184,59 @@ If this case happens, the implicit filters are regenerated and the function is c
...
+              query: agentId,
+              type: 'phrase',
+            },
+          },
+        },
+        $state: { store: 'appState', isImplicit: true },
+      };
+    }
+    return undefined;
+  }
```
Comment (yenienserrano) [inline plugins/main/public/components/common/modules/modules-helper.js:241]: is it still necessary with the dataSource?

File: plugins/main/public/components/overview/threat-hunting/config/index.tsx
Comment (yenienserrano) [inline plugins/main/public/components/overview/threat-hunting/config/index.tsx:1]: The columns of the events tab in the other modules are in an events folder.

File: plugins/main/public/components/common/modules/modules-helper.js (hunk 5 above)
Comment (jbiset) [reply plugins/main/public/components/common/modules/modules-helper.js:241]: Removed

File: plugins/main/public/components/common/modules/modules-defaults.tsx (hunk 4 above)
Comment (jbiset) [reply plugins/main/public/components/common/modules/modules-defaults.tsx:49]: Threat Hunting does not have a rule group filter, so the normal alert filter works

File: plugins/main/public/components/overview/threat-hunting/config/index.tsx
Comment (jbiset) [reply plugins/main/public/components/overview/threat-hunting/config/index.tsx:1]: Changed Threat Hunting columns file name

**Comment by yenienserrano** [comment]
## Test
Go to Threats Hunting and check the following:
- [ ] Each visualization, if applicable, must have interaction so that it adds the corresponding filter(s) upon clicking.
- [ ] The visualizations have to be updated according to the filters applied in the searchbar.
- [ ] The visualizations have to be updated when a search is performed in the searchbar.
- [ ] If there are no results, the corresponding message must appear that there are no results and the visualizations should not be rendered.
- [ ] If there is SampleData, the corresponding SampleData message must appear.
- [ ] If an agent is pinned, the views must be updated and changed to the agent view, if applicable. 
- [ ] Check filter behavior with respect to `hide alerts`, `allow agents` and filter order.

File: plugins/main/public/components/common/modules/modules-helper.js [hunk 6]
```diff
@@ -2,6 +2,8 @@ import { getAngularModule, getDataPlugin } from '../../../kibana-services';
 import { AppState } from '../../../react-services/app-state';
 import { FilterHandler } from '../../../utils/filter-handler';
 import { VULNERABILITY_IMPLICIT_CLUSTER_MODE_FILTER } from '../../../../common/constants';
+import { useFilterManager } from '../hooks';
+import { FilterStateStore } from '../../../../../../src/plugins/data/common';
```
[the remark this answers is not in the document]
Comment (jbiset) [reply plugins/main/public/components/common/modules/modules-helper.js]: Deleted unused imports

**Comment by JuanGarriuz** [comment]
## Test
Go to Threats Hunting and check the following:
- [ ] Each visualization, if applicable, must have interaction so that it adds the corresponding filter(s) upon clicking.


https://github.com/wazuh/wazuh-dashboard-plugins/assets/124377319/60c88b33-494c-4e88-8df1-5339d079af2d



- [ ] The visualizations have to be updated according to the filters applied in the searchbar.

image



- [ ] The visualizations have to be updated when a search is performed in the searchbar.

image


- [ ] If there are no results, the corresponding message must appear that there are no results and the visualizations should not be rendered.

image


- [ ] If there is SampleData, the corresponding SampleData message must appear.

image


- [ ] If an agent is pinned, the views must be updated and changed to the agent view, if applicable. 

image


- [ ] Check filter behavior with respect to `hide alerts`, `allow agents` and filter order. 

image
image

**Review by yenienserrano — DISMISSED** [review]
LGTM

**Review by JuanGarriuz — APPROVED** [review]

**Review by asteriscos — APPROVED** [review]

**Pull request merged**